projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobEphemeralStorageIT.java

```diff
@@ -0,0 +1,380 @@
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.user;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Slf4j
+@Import({DataJobDeploymentCrudIT.TaskExecutorConfig.class})
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = ControlplaneApplication.class)
+public class DataJobEphemeralStorageIT extends BaseIT {
+
+  private static final String TEST_JOB_NAME =
+      "ephemeral-storage-test-" + UUID.randomUUID().toString().substring(0, 8);
+  private static final Object DEPLOYMENT_ID = "testing-ephemeral-storage";
+  private final ObjectMapper objectMapper =
+      new ObjectMapper()
+          .registerModule(new JavaTimeModule()); // Used for converting to OffsetDateTime;
+
+  @AfterEach
+  public void cleanUp() throws Exception {
+    // delete job
+    mockMvc
+        .perform(
+            delete(
+                    String.format(
+                        "/data-jobs/for-team/%s/jobs/%s/sources", TEST_TEAM_NAME, TEST_JOB_NAME))
+                .with(user("user")))
+        .andExpect(status().isOk());
+
+    // Execute delete deployment
+    mockMvc
+        .perform(
+            delete(
+                    String.format(
+                        "/data-jobs/for-team/%s/jobs/%s/deployments/%s",
+                        TEST_TEAM_NAME, TEST_JOB_NAME, DEPLOYMENT_ID))
+                .with(user("user"))
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isAccepted());
+  }
+
+  @BeforeEach
+  public void setup() throws Exception {
+    String dataJobRequestBody = getDataJobRequestBody(TEST_TEAM_NAME, TEST_JOB_NAME);
+
+    // Execute create job
+    mockMvc
+        .perform(
+            post(String.format("/data-jobs/for-team/%s/jobs", TEST_TEAM_NAME))
+                .with(user("user"))
+                .content(dataJobRequestBody)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isCreated())
+        .andExpect(
+            header()
+                .string(
+                    HttpHeaders.LOCATION,
+                    lambdaMatcher(
+                        s ->
+                            s.endsWith(
+                                String.format(
+                                    "/data-jobs/for-team/%s/jobs/%s",
+                                    TEST_TEAM_NAME, TEST_JOB_NAME)))));
+  }
+
+  @Test
+  public void testEphemeralStorageJob() throws Exception {
```

Comment:
refactored and moved duplicated code to BaseIT

Reply:
I don't see the refactoring .... 

Btw I am fine if you decide to do this refactoring in separate PR as long as we have an agreement that it will be done with priority.

Reply:
I cut it as per Miro's suggestion since I was having issues with getting it to work and getting this merged is higher on the priority list.

Reply:
As I said I think it's ok to merge it as long as we have an agreement on whether it will be refactored or not. 

In any case yes, it's better that purely refactoring changes are done separated from functional changes as it's recommended here - https://github.com/vmware/versatile-data-kit/wiki/How-to-prepare-a-new-PR#tackle-non-functional-or-refactoring-changes-first (the recommendation is that refactoring change is first but the spirit of it is the same).

Reply:
Just in case it's not clear. I am looking from you if you agree or not. if not why not and we can discuss (I might be wrong). 
 If yes, let's create an issue for that and plan to fix it.

Reply:
I will make sure to refactor it, I've made a note of it.

Reply:
Story: https://github.com/vmware/versatile-data-kit/issues/1330